rest/admin-v2/rest/src/main/java/org/keycloak/admin/api/client/DefaultClientApi.java

```diff
@@ -104,8 +104,8 @@ public void deleteClient() {
         clientResource.deleteClient();
     }
 
-    static void validateUnknownFields(ClientRepresentation rep) {
-        if (!rep.getAdditionalFields().isEmpty()) {
+    static void validateUnknownFields(BaseClientRepresentation rep) {
+        if (rep.getAdditionalFields().keySet().stream().anyMatch(k -> !k.equals(BaseClientRepresentation.DISCRIMINATOR_FIELD))) {
```

Comment:
If having the protocol field not be part of the schema is causing it to be parsed into the additionalFields, perhaps we should just have it as a proper field.